src/lib.rs

```diff
@@ -143,12 +156,49 @@ impl<'a> EbpfVmMbuff<'a> {
     /// ];
     ///
     /// // Instantiate a VM.
-    /// let mut vm = rbpf::EbpfVmMbuff::new(prog1);
+    /// let mut vm = rbpf::EbpfVmMbuff::new(Some(prog1));
     /// vm.set_prog(prog2);
     /// ```
     pub fn set_prog(&mut self, prog: &'a [u8]) {
-        verifier::check(prog);
-        self.prog = prog;
+        (self.verifier)(prog);
+        self.prog = Some(prog);
+    }
+
+    /// Sets a new verifier function
+    ///
+    /// # Panics
+    ///
+    /// The simple verifier may panic if it finds errors in the eBPF program at load time.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use rbpf::ebpf;
+    ///
+    /// // simple verifier
+    /// fn verifier(prog: &[u8]) -> bool {
+    ///     let last_insn = ebpf::get_insn(prog, (prog.len() / ebpf::INSN_SIZE) - 1);
+    ///     if last_insn.opc != ebpf::EXIT {
+    ///         panic!("[Verifier] Error: program does not end with “EXIT” instruction");
+    ///     }
+    ///     true
+    /// }
+    ///
+    /// let prog1 = &[
+    ///     0xb7, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00, 0x00, // mov r0, 0
```

Comment:
You copied and pasted that line and I can't blame you. There's an error from the line you copied, the `0x02` should be `0x00`. I fixed several occurrences of this mistake yesterday.